### DISCUSSION TITLE
To conduct test using SEB with webcam support.

### DISCUSSION THREAD

**Discussion by sierratec** [SafeExamBrowser/seb-win-refactoring#87] [Q&A]
Hi,

We are using Moodle 3.9 and we would like to conduct test using SEB with webcam support.  We tried SEB 3.1 and this is working. We are able to download the config file an clicking on which able to open the test. Since SEB 3.1 does not have the support to webcam, please suggest how to open the test session (either the web URL or the config file) in Chrominium.

Alternatively we tried Moodle 3.9 with SEB 2.4. Here if we try to open the test session just a blank screen only opens. Could you please suggest.

We are checking this in Windows 10

Thanks

**Comment by dbuechel** [comment]
You'd add Chrominimum as allowed third-party application and then set the URL as command-line parameter in the configuration: https://safeexambrowser.org/windows/win_usermanual_en.html#PermittedProcessesSection. All configuration options of Chrominimum are explained in its README: https://github.com/SafeExamBrowser/Chrominimum#readme

**Comment by sierratec** [comment]
Hi,

Thanks for your reply.

To make  ‘Chrominium’ to work correctly, we checked and noticed that we need to have .NET framework and Visual C++ installed in the m/c. As we cannot ask the end users who come for taking the test to install all these, we planned to go with SEB 2.4.

We installed SEB 2.4 and the configuration related screen shots are here: http://www.sierratec.com/downloads/ref-260521.docx

We have following queries
1.	After this if we click the SebClientSettings.seb file in C:\Users\xxxx\AppData\Roaming\SafeExamBrowser, I am getting following alert “New settings have been saved, they will be used when you start SEB next time again. Do you want to start working with SEB or quit for now?” .This alert appears even though we save the settings. Please suggest how to fix this.
2.	We have checked ‘Allow video capture’ and ‘Allow audio capture (microphone)’ . Please suggest where these will be stored.

Please help

Thanks

**Comment by dbuechel** [comment]
The setup bundle automatically installs the runtime dependencies, there is no need for your end users to install these manually. SEB 2.x isn't actively developed anymore, we highly recommend using SEB 3.x whenever possible.

1. Once a client computer has been configured with a configuration, there is no need to navigate to it and start it by double-clicking. You can simply start SEB e.g. from the start menu and it will use the configuration stored in the user's APPDATA directory.
2. This simply allows the browser to access camera and microphone, there won't be any recordings stored by activating these options.

**Comment by sierratec** [comment]
Hi,

Thanks for your reply

Since SEB  3.1 is the latest version, we will use SEB  3.1

We installed SEB_3.1.1.250_SetupBundle.

Could you please clarify below queries
1) We installed Chrominimum_1.1.1.30_SetupBundle. As you suggested we understand that this will automatically install the runtime dependencies  like .NET framework and Visual C++. Please confirm
2) While opening SEB config tools we got ‘Default Prohibited Processes Found’ alert. We have provided the screen shots in page No. 1 in the attachment . How can we fix this.
3) As you suggested we are not able to start SEB from start menu. We have provided the screen shots in page No. 2 in the attachment . How can we fix this
4) We are able to open a web url via command line. We have provided the screen shot in page No. 3 in the attachment. Do we have the possibility to open a web url by clicking the seb file instead of going to command prompt

Screen shot: http://www.sierratec.com/downloads/ref-270521.docx

**Reply by dbuechel** [reply]
1. Exactly, that is correct.
2. This is only a message, not an error. We improved the handling of this message in the upcoming version 3.2, for the time being you can simply ignore it.
3. SEB is registered as "Safe Exam Browser", thus you'd either need to navigate to its dedicated place in the application list of the start menu or search it by typing "Safe ...".
4. The idea of Chrominimum is to integrate it within SEB, i.e. configure it as permitted process in the configuration tool (under tab "Applications", see https://safeexambrowser.org/windows/win_usermanual_en.html#PermittedProcessesSection). For each permitted process, you can furthermore add command-line parameters under "Arguments", so you'd add an URL there if you want that Chrominimum opens a certain resource.

The upcoming version 3.2 will be able to access microphone and webcam directly within the main browser, as issue #3 has been resolved. We project a release this week, if all goes well, so you might want to wait for version 3.2 and then setup your assessment scenario accordingly.

**Reply by sierratec** [reply]
Hi,

Thanks for your support and detailed reply.

Could you please confirm whether the end user has to install 'Chrominimum_1.1.1.30_SetupBundle.exe' (162MB) to take the test or is there any smaller version

**Reply by dbuechel** [reply]
No, you won't need to install and use Chrominimum with SEB 3.2.

**Reply by sierratec** [reply]
Noted. Thanks for your reply.

**Reply by sierratec** [reply]
Hi,
We installed SEB 3.2.
1.	We are able to open a web url using ‘SebClientSettings’ file located in the path 	C:\Users\xxxx\AppData\Roaming\SafeExamBrowser 
2.	In the SEB Configuration Tool, we are not  able to find the provision to enable web cam. Please suggest how can we enable web cam. We checked both ‘User Interface’ and ‘Security’ tabs
Thank you

**Reply by dbuechel** [reply]
The webcam and microphone are automatically enabled, they don't need to be enabled in the configuration.

**Reply by sierratec** [reply]
Hi,

I am using Windows 10 and in my laptop my camera is not enabled. In the same m/c only I installed SEB. Please suggest whether I need to use a different m/c for the test.

Also please suggest how can we check the test session from admin side like the candidates web cam captures.

One more info:
If I check my webcam using webcam js sample (https://makitweb.com/demos/how-to-capture-picture-from-webcam-with-webcam-js-1) and my web cam is getting enabled. So I understand that in my m/c there is no issues with my webcam

Thank you

**Reply by sierratec** [reply]
Hi,

Also , could you please confirm the below steps
1) Once we setup the web url in SEB configuration tool, we need to share the ‘SebClientSettings.seb' (available in the path C:\Users\xxxx\AppData\Roaming\SafeExamBrowser) to the end user who is going to take the test
2) Then the end user (who is going to take the test) has to install SEB lite. 
If yes, to the above then
3) How can we create and share a downloadable link for this SEB like to the end user like the below. If any pricing involved for this. If yes. How much
   a)  https://www.icsi.edu/media/webmodules/SEB_Instructions_Manual.pdf -> Step 1
   b)   https://acoe.annauniv.edu/download_forms/student_forms/A2%20-%20Candidate%20User%20Manual%20-%20Mobile%20App.pdf -> Page No. 4
Thank you

**Reply by dbuechel** [reply]
Try https://test.webrtc.org/ within SEB to verify whether microphone and camera access work. SEB Lite is not a product we develop, you'd need to ask the developers of SEB Lite how to distribute and configure their product.

**Reply by sierratec** [reply]
Hi 
1.	If we configure this url  https://test.webrtc.org/ using SEB Configuration Tool -> General -> Start URL and then if we open this using 'SebClientSettings.seb' the website is opening in SEB and the webcam is also getting enabled
2.	But if we configure and open a Moodle quiz url similar to the above, the website is opening in SEB but the webcam is not getting enabled. 
3.	Also we tried to setup the Moodle quiz as in this reference  https://docs.moodle.org/311/en/Safe_Exam_Browser. Here the website is opening in SEB but the webcam is not getting enabled. 
Could you please suggest how can we enable webcam
Thank you

**Reply by dbuechel** [reply]
Your webapplication / LMS needs to have a functionality or plugin which enables e.g. remote proctoring. In other words, https://test.webrtc.org enables the webcam because that website has JavaScript code which explicitly accesses the webcam. Thus you'd either need to install a plugin for remote proctoring in your LMS or manually develop a feature which accesses the webcam and provides remote proctoring functionality.

**Reply by sierratec** [reply]
Hi,
We are in the process of checking an applicable Moodle plugin.
Also Step 2 of this link https://mahephd.azurewebsites.net/pdf/SEB%20and%20Mock%20PhD%20MET%202020%20Instructions.pdf allows us to download MAHE-SEBLite-Chrome.msi. I googled and couldn't find the developers of SEB Lite. If you know the details could you please share.
Thank you
